tests/Controller/ExportControllerTest.php

```diff
@@ -228,4 +214,189 @@ public function testCorrectTranslatedProductIsExported(): void
         $this->assertSame(1, (int)$parsedResponse->items->attributes()->count);
         $this->assertSame('FINDOLOGIC Product DE', $parsedResponse->items->item->names->name->__toString());
     }
+
+    /**
+     * Test ensures that the Shopware Router generates the product URLs based on the current language.
+     * This special case applies if a Sales Channel has multiple languages and the export is called from
+     * within a different Sales Channel. E.g.
+     *
+     * Sales Channel "Germany" can be accessed with URL https://some-shop.de, therefore the configured export
+     * URL is set to it. Now there is a second domain for "United Kingdom", which can be accessed with the
+     * URL https://some-shop.co.uk. This test ensures that the products will use the UK domain, when calling
+     * the URL of the German Sales Channel with the shopkey of the United Kingdom Sales Channel:
+     * https://some-shop.de/findologic?shopkey=from-united-kingdom => https://shome-shop.co.uk/detail/...
+     *
+     * Runs in a separate process to not interfere with other tests.
+     * @runInSeparateProcess
```

Comment:
Why is this necessary? Because deletion of seo urls?

Reply:
I couldn't pin down the exact reason, but the test runs fine when executed in a single scoped thread and I think that it may be related to Shopware, since they basically reset the database on each test and somehow that one product can then be not found. But then again, we are using a fake client to send the request. I think that even if it would pass (which it doesn't), going for the separate process makes this more stable overall.